operators/src/source/gdal_source.rs

```diff
@@ -385,7 +388,18 @@ where
             &output_bounds
         );
 
-        let dataset_result = GdalDataset::open(&dataset_params.file_path);
+        let dataset_result = if let Some(open_options) = dataset_params.gdal_open_options.as_ref() {
+            let options = open_options.iter().map(String::as_str).collect::<Vec<_>>();
+            GdalDataset::open_ex(
+                &dataset_params.file_path,
+                DatasetOptions {
+                    open_options: Some(options.as_slice()),
+                    ..DatasetOptions::default()
+                },
+            )
+        } else {
+            GdalDataset::open(&dataset_params.file_path)
+        };
```

Comment:
We could use only one open_ex call and pass `None` or an empty vector. This would mean less paths :shrug:.

Reply:
93573df8a3d72df781b957bc4249b90c9b63a5d3